awx/ui_next/src/screens/Project/shared/ProjectForm.jsx

```diff
@@ -30,19 +30,19 @@ const ScmTypeFormRow = styled(FormRow)`
   padding: 24px;
 `;
 
-function ProjectForm(props) {
-  const { project, handleCancel, handleSubmit, i18n } = props;
+function ProjectForm({ project, ...props }) {
+  const { i18n, handleCancel, handleSubmit } = props;
+  const { summary_fields = {} } = project;
   const [contentError, setContentError] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
-  const [organization, setOrganization] = useState(null);
+  const [organization, setOrganization] = useState(
+    summary_fields.organization || null
+  );
+  const [scmSubFormState, setScmSubFormState] = useState(null);
   const [scmTypeOptions, setScmTypeOptions] = useState(null);
-  const [scmCredential, setScmCredential] = useState({
-    typeId: null,
-    value: null,
-  });
-  const [insightsCredential, setInsightsCredential] = useState({
-    typeId: null,
-    value: null,
+  const [credentials, setCredentials] = useState({
+    scm: { typeId: null, value: null },
+    insights: { typeId: null, value: null },
   });
 
   useEffect(() => {
```

Comment:
this useEffect method is getting pretty long. Is it possible/worthwhile to pull out the data fetching into a helper method that returns credentials for `setCredentials`?

Reply:
Since I was only using those functions inside of the effect, I moved them directly into the effect to avoid transitive dependencies.  We could move those requests outside of `useEffect` if it’ll improve readability.